models/factor.go

```diff
@@ -44,31 +56,32 @@ func (authMethod AuthenticationMethod) String() string {
 }
 
 type Factor struct {
-	ID           uuid.UUID `json:"id" db:"id"`
-	User         User      `json:"-" belongs_to:"user"`
-	UserID       uuid.UUID `json:"-" db:"user_id"`
-	CreatedAt    time.Time `json:"created_at" db:"created_at"`
-	UpdatedAt    time.Time `json:"updated_at" db:"updated_at"`
-	Status       string    `json:"status" db:"status"`
-	FriendlyName string    `json:"friendly_name,omitempty" db:"friendly_name"`
-	Secret       string    `json:"-" db:"secret"`
-	FactorType   string    `json:"factor_type" db:"factor_type"`
+	ID           uuid.UUID   `json:"id" db:"id"`
+	User         User        `json:"-" belongs_to:"user"`
+	UserID       uuid.UUID   `json:"-" db:"user_id"`
+	CreatedAt    time.Time   `json:"created_at" db:"created_at"`
+	UpdatedAt    time.Time   `json:"updated_at" db:"updated_at"`
+	Status       string      `json:"status" db:"status"`
+	FriendlyName string      `json:"friendly_name,omitempty" db:"friendly_name"`
+	Secret       string      `json:"-" db:"secret"`
+	FactorType   string      `json:"factor_type" db:"factor_type"`
+	Challenge    []Challenge `json:"challenges,omitempty" has_many:"challenges"`
```

Comment:
Does it make sense to return these in JSON?

Reply:
I feel like if we expose this information, we'd be leaking unnecessary information to _other_ sessions. So it's fine to learn of a challenge by calling it on the device, but not learn about other challenges on other devices.